tachyon/crypto/hashes/sponge/poseidon2/poseidon2_config.h

```diff
@@ -63,6 +64,19 @@ struct Poseidon2Config : public PoseidonConfigBase<Params> {
       : PoseidonConfigBase<Params>(std::move(base)),
         internal_diagonal_minus_one(std::move(internal_diagonal_minus_one)) {}
 
+  constexpr static Poseidon2Config CreateDefault() {
+    if constexpr (Params::kInternalMatrixVendor == Poseidon2Vendor::kPlonky3) {
+      return Create(GetPoseidon2InternalShiftArray<Params>());
+    } else {
+      return Create(GetPoseidon2InternalDiagonalArray<Params>());
+    }
+  }
```

Comment:
Even though `CreateDefault()` for `Plonky3` gives us `GetPoseidon2InternalShiftArray`, it appears as though we choose pretty equally between `GetPoseidon2InternalShiftArray` (which we use CreateDefault()) and `GetPoseidon2InternalDiagonalArray` (which we create config and run Create()) when using a plonky3 internal matrix. Perhaps a bool parameter input to to denote `GetPoseidon2InternalShiftArray` or `GetPoseidon2InternalDiagonalArray` might be a better solution (though our Horizen instances appear to only use `GetPoseidon2InternalDiagonalArray`)